firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -880,15 +891,16 @@ def pbs_for_set(document_path, document_data, option):
         or two ``Write`` protobuf instances for ``set()``.
     """
     transform_paths, actual_data = remove_server_timestamp(document_data)
-
     update_pb = write_pb2.Write(
         update=document_pb2.Document(
             name=document_path,
             fields=encode_dict(actual_data),
         ),
     )
     if option is not None:
```

Comment:
This should only be true for `option.merge = true`

Reply:
`field_paths` is ignored in all other options, and I had merge take and ignore any argument from what we discussed earlier.  Do you only want it to work for `merge=True` and ignore all other values?

Reply:
@schmidt-sebastian I have passed all of the conformance tests locally.  I still have cleanups to do though.

Do you want me to do the entire thing in one new PR or would you like to merge this first and then work on the diffs from this one?